### PR TITLE
Windows editor start maximized if exited in this state

### PR DESCRIPTION
**Pull request by rsredsq**
#511

### CODE REVIEW

**Comment by JoshEngebretson** [comment]
Mr. Squirrel, this is looking great and will be a nice improvement.  I'll have a closer look once we're back in the office.  Happy holidays! :christmas_tree:

File: Script/AtomicEditor/editor/Editor.ts [hunk 1]
```diff
@@ -170,7 +200,7 @@ class Editor extends Atomic.ScriptObject {
     }
 
     exit() {
-        Preferences.getInstance().write();
+        //Preferences.getInstance().write();
```
Comment (JoshEngebretson) [inline Script/AtomicEditor/editor/Editor.ts]: Dead code?

**Comment by JoshEngebretson** [comment]
Ok, I looked this over and other than one note it looks really good.  It just needs some testing now.

File: Source/Atomic/Engine/Engine.cpp [hunk 2]
```diff
@@ -379,6 +379,9 @@ bool Engine::Initialize(const VariantMap& parameters)
         ))
             return false;
 
+        if (GetParameter(parameters, "WindowMaximized", false) == true)
+            graphics->Maximize();
```
Comment (JoshEngebretson) [inline Source/Atomic/Engine/Engine.cpp]: This should be factored into the SetMode call immediately above:

https://github.com/rsredsq/AtomicGameEngine/blob/RED-EDITOR-511/Source/Atomic/Engine/Engine.cpp#L370

That way the window isn't created and then immediately resized for maximize

**Comment by LaraEngebretson** [comment]
This issue occurs on OSX only:

* Move the editor to 2nd monitor
* Double-click the title bar to maximize the editor window
* Move editor back to primary monitor
* Close the editor
* Relaunch the editor

Issue: the editor reopens on the 2nd monitor

File: Source/Atomic/Engine/Engine.cpp (hunk 2 above)
Comment (rsredsq) [inline Source/Atomic/Engine/Engine.cpp]: ok

**Comment by rsredsq** [comment]
@LaraEngebretson thanks! good catch ;)

**Pull request merged**